merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/http/MerlinBindings.java

```diff
@@ -200,12 +207,14 @@ private void getSimulationResults(final Context ctx) {
     }
   }
 
-  private void getResourceSamples(final Context ctx) {
+  private void getResourceSamples(final Context ctx) throws Unauthorized {
     try {
-      final var planId = parseJson(ctx.body(), hasuraPlanActionP).input().planId();
+      final var body = parseJson(ctx.body(), hasuraPlanActionP);
+      final var planId = body.input().planId();
 
-      final var resourceSamples = this.simulationAction.getResourceSamples(planId);
+      this.checkPlanPermissions("resource_samples", body.session(), planId);
```

Comment:
Is it possible to encode these `action_permission` keys like `"resource_samples"` as an enum to better capture the possible options in our Java code? Otherwise, it seems the only place these are listed are this [github discussion](https://github.com/NASA-AMMOS/aerie/discussions/983#discussioncomment-6257146)